### PR TITLE
Plugin

### PR DESCRIPTION
**Pull request by jmfayard**
Plugin documentation: 
https://github.com/LouisCAD/Splitties/tree/plugin/plugin

Run ` ./gradlew :plugin:publishToMavenLocal`

Run `$ ./gradlew migrateToAndroidX` in the project that needs to be migrated

Running `:plugin:publishPlugins` will publish a version to https://plugins.gradle.org/plugin/com.louiscad.splitties
This in turns will allow us to test it more easily in other projects. 

What about the version? I considered three alternatives:

- using an exact version number -- but it will never be "the right version" for all dependencies and I find it annyoing
- using `_` -- but that makes the Android Sync fails, so no
- using `+` -- that works and only needs to be explained

I kept the version for Splitties, which is a special case, and for AndroidX (just for now)

Caveat: [Android Studio does not seem to support auto-completion for build.gradle](https://github.com/LouisCAD/Splitties/issues/220)

----

<strike>The path to the CSV file is hardcoded.  It should be possible to use the CSV from the `resources` folder but I couldn't find the right syntax for this. https://github.com/LouisCAD/Splitties/blob/plugin/plugin/src/main/kotlin/com/louiscad/splitties/MigrateAndroidxTask.kt#L14-L16</strike>

### CODE REVIEW

File: plugin/src/main/kotlin/com/louiscad/splitties/PluginConfig.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.louiscad.splitties
+
+
+object PluginConfig {
+    const val GROUP = "com.louiscad.splitties"
+    const val GRADLE_PROPERTY = "version.$GROUP"
+    const val SPLITTIES_VERSION = "2.1.1"
```
Comment (LouisCAD) [inline plugin/src/main/kotlin/com/louiscad/splitties/PluginConfig.kt]: Shouldn't this point to the latest alpha release?

File: plugin/src/main/kotlin/com/louiscad/splitties/Square.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+/**
+ * The actual dependency version comes from `gradle.properties`
+ * from either `version.$NAME=xxx` or `version.com.squareup.okhttp=xxx` or `version.com.squreup.retrofit2=xxxx`
+ **/
+object Square {
+    private const val okHttpVersion = "3.12.0"
+    const val okHttp = "com.squareup.okhttp3:okhttp:$okHttpVersion"
+    const val okHttpLoggingInterceptor = "com.squareup.okhttp3:logging-interceptor:$okHttpVersion"
+    private const val retrofitversion = "2.5.0"
```
Comment (LouisCAD) [inline plugin/src/main/kotlin/com/louiscad/splitties/Square.kt]: Can be updated to 2.6.2

File: plugin/src/main/kotlin/com/louiscad/splitties/PluginConfig.kt (hunk 1 above)
Comment (jmfayard) [reply plugin/src/main/kotlin/com/louiscad/splitties/PluginConfig.kt]: done

File: plugin/src/main/kotlin/com/louiscad/splitties/Square.kt (hunk 2 above)
Comment (jmfayard) [reply plugin/src/main/kotlin/com/louiscad/splitties/Square.kt]: replaced with `+`, see PR description

**Pull request merged**